src/NLog/Targets/FileTarget.cs

```diff
@@ -237,8 +237,11 @@ public Layout FileName
 
                 fileName = value;
 
-                RefreshFileArchive();
-                RefreshArchiveFilePatternToWatch();
+                if (IsInitialized)
+                {
+                    RefreshFileArchive();
+                    RefreshArchiveFilePatternToWatch();
```

Comment:
I think it's more clear (for the other properties), to keep the check in the setter, but I will double check the other properties.